.github/workflows/test.yml

```diff
@@ -0,0 +1,147 @@
+        sudo wget https://github.com/containernetworking/plugins/releases/download/v1.1.1/cni-plugins-linux-amd64-v1.1.1.tgz
+        sudo tar Cxzvf /opt/cni/bin cni-plugins-linux-amd64-v1.1.1.tgz
+
+    - name: Setup minikube
+      if: matrix.use-existing-cluster
+      uses: medyagh/setup-minikube@latest
+      with:
+        driver: docker
+        container-runtime: containerd
+        cni: flannel
+
+    - name: Install rqlite cluster
+      if: matrix.use-existing-cluster
+      run: |
+        curl -s https://raw.githubusercontent.com/rqlite/kubernetes-configuration/master/service.yaml -o rqlite-service.yaml
+        kubectl apply -f rqlite-service.yaml
+
+        kubectl apply -f - <<EOF
+          apiVersion: apps/v1
+          kind: StatefulSet
+          metadata:
+            name: rqlite
+          spec:
+            selector:
+              matchLabels:
+                app: rqlite # has to match .spec.template.metadata.labels
+            serviceName: rqlite-svc-internal
+            replicas: 3
+            podManagementPolicy: "Parallel"
+            template:
+              metadata:
+                labels:
+                  app: rqlite # has to match .spec.selector.matchLabels
+              spec:
+                terminationGracePeriodSeconds: 5
+                containers:
+                - name: rqlite
+                  image: rqlite/rqlite
+                  imagePullPolicy: IfNotPresent
+                  args: ["-disco-mode=dns","-disco-config={\"name\":\"rqlite-svc-internal\"}","-bootstrap-expect","3", "-join-interval=1s", "-join-attempts=120"]
+                  ports:
+                  - containerPort: 4001
+                    name: rqlite
+                  readinessProbe:
+                    httpGet:
+                      scheme: HTTP
+                      path: /readyz
+                      port: 4001
+                    periodSeconds: 5
+                    timeoutSeconds: 2
+                    initialDelaySeconds: 2
+                  livenessProbe:
+                    httpGet:
+                      scheme: HTTP
+                      path: /readyz?noleader
+                      port: rqlite
+                    initialDelaySeconds: 2
+                    timeoutSeconds: 2
+                    failureThreshold: 3
+                  volumeMounts:
+                  - name: rqlite-file
+                    mountPath: /rqlite/file
+                volumes:
+                - name: rqlite-file
+                  emptyDir:
+        EOF
+
+    - name: Run unit tests that depend on existing k8s cluster
+      if: matrix.use-existing-cluster
+      env:
+        USE_EXISTING_CLUSTER: ${{ matrix.use-existing-cluster }}
+      # set label filter 'k8s' to ignore the test cases that don't deponed on really k8s cluster
+      run: ginkgo run --label-filter 'k8s' controllers/
+
+    - if: failure()
+      run: kubectl get pods -A
+
+    - if: failure()
+      run: kubectl get event -n kube-system
```

Comment:
I think this is needless

Reply:
I think it can help us to analysis the reason if occurred failure, it has no effect here